extensions-core/druid-pac4j/src/main/java/org/apache/druid/security/pac4j/Pac4jFilter.java

```diff
@@ -48,7 +52,29 @@ public class Pac4jFilter implements Filter
 {
   private static final Logger LOGGER = new Logger(Pac4jFilter.class);
 
-  private static final HttpActionAdapter<String, JEEContext> NOOP_HTTP_ACTION_ADAPTER = (HttpAction code, JEEContext ctx) -> null;
+  private static final HttpActionAdapter<String, JEEContext> HTTP_ACTION_ADAPTER = (HttpAction action, JEEContext context) -> {
+    if (action instanceof RedirectionAction) {
+      int code = action.getCode();
+      HttpServletResponse response = context.getNativeResponse();
+      response.setStatus(code);
+      if (action instanceof WithLocationAction) {
+        WithLocationAction withLocationAction = (WithLocationAction) action;
+        context.setResponseHeader("Location", withLocationAction.getLocation());
+      } else if (action instanceof WithContentAction) {
+        WithContentAction withContentAction = (WithContentAction) action;
+        String content = withContentAction.getContent();
+        if (content != null) {
+          try {
+            response.getWriter().write(content);
+          }
+          catch (IOException var8) {
```

Comment:
curious where this `var8` came from. Did we copy this code from somewhere? If so we should clarify where this came from.

Reply:
yeah, I copied it from the JEEHttpActionAdapter. I have updated it to use JEEHttpActionAdapter instead of creating a new one.